### PR TITLE
[JBIDE-22442] Port Forwarding wizard: When managing forwarding ports, UI is unresponsive when playing with find free local ports

### PR DESCRIPTION
**Pull request by jeffmaury**
Suppress TCP connection establishment to checked port

Signed-off-by: Jeff MAURY <jmaury@redhat.com>

### CODE REVIEW

File: plugins/org.jboss.tools.openshift.core/src/org/jboss/tools/openshift/internal/core/portforwarding/PortForwardingUtils.java [hunk 1]
```diff
@@ -53,15 +53,11 @@ public static boolean isPortInUse(int port) {
 			return false;
 		}
 		try (ServerSocket socket = new ServerSocket(port)){
-			//not in use apparently
+			//not in use
+		    return false;
```
Comment (adietish) [inline plugins/org.jboss.tools.openshift.core/src/org/jboss/tools/openshift/internal/core/portforwarding/PortForwardingUtils.java]: obvious, +1

File: plugins/org.jboss.tools.openshift.core/src/org/jboss/tools/openshift/internal/core/portforwarding/PortForwardingUtils.java [hunk 2]
```diff
@@ -53,15 +53,11 @@ public static boolean isPortInUse(int port) {
...
 		} catch (IOException e) {
 			return true;//in use
 		}
-		try (Socket socket = new Socket("localhost", port)) {
-			return true; // in use
-		} catch (IOException e) {
-			return false;
-		}
```
Comment (adietish) [inline plugins/org.jboss.tools.openshift.core/src/org/jboss/tools/openshift/internal/core/portforwarding/PortForwardingUtils.java]: this code looks obscure, trying to allocate a port twice, once with host, once without. Any idea why this was done? Is this maybe something that's required on specific platforms (Mac?). 
I'd suggest that we test this on all 3 platforms before merging.

Comment (jeffmaury) [inline plugins/org.jboss.tools.openshift.core/src/org/jboss/tools/openshift/internal/core/portforwarding/PortForwardingUtils.java]: Yes that's what Xavier told me (mentioning Mac) so I tested it on Mac and Win10

**Comment by adietish** [comment]
I tested this on Linux and it works flawlessly.
@fbricon can you test this on Mac please?

File: plugins/org.jboss.tools.openshift.core/src/org/jboss/tools/openshift/internal/core/portforwarding/PortForwardingUtils.java (hunk 2 above)
Comment (adietish) [inline plugins/org.jboss.tools.openshift.core/src/org/jboss/tools/openshift/internal/core/portforwarding/PortForwardingUtils.java]: @jeffmaury ok, great. Merging then.

File: plugins/org.jboss.tools.openshift.core/src/org/jboss/tools/openshift/internal/core/portforwarding/PortForwardingUtils.java (hunk 1 above)
Comment (scabanovich) [inline plugins/org.jboss.tools.openshift.core/src/org/jboss/tools/openshift/internal/core/portforwarding/PortForwardingUtils.java]: @adietish , the previous logic was implemented when doing JBIDE-21847. It seemed then that the check with ServerSocket was not always sufficient. @fbricon , could you please try to remember why we kept both checks?

**Pull request merged**